samples/src/Microsoft.Azure.IIoT.App/src/Services/Publisher.cs

```diff
@@ -0,0 +1,104 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.Azure.IIoT.App.Services {
+    using Microsoft.AspNetCore.Authentication.AzureAD.UI.Pages.Internal;
+    using Microsoft.Azure.IIoT.App.Data;
+    using Microsoft.Azure.IIoT.OpcUa.Api.Publisher;
+    using Microsoft.Azure.IIoT.OpcUa.Api.Publisher.Models;
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Browser code behind
+    /// </summary>
+    public class Publisher {
+
+        /// <summary>
+        /// Create browser
+        /// </summary>
+        /// <param name=""></param>
+        public Publisher(IPublisherServiceApi publisherService) {
+            _publisherService = publisherService;
+        }
+
+        /// <summary>
+        /// PublishedAsync
+        /// </summary>
+        /// <param name="endpointId"></param>
+        /// <returns>PublishedNode</returns>
+        public async Task<PagedResult<PublishedNode>> PublishedAsync(string endpointId) {
+            var pageResult = new PagedResult<PublishedNode>();
+            try {
+                string continuationToken = string.Empty;
+                do {
+                    var result = await _publisherService.NodePublishListAsync(endpointId, continuationToken);
+                    continuationToken = result.ContinuationToken;
+
+                    if (result.Items != null) {
+                        foreach (var item in result.Items) {
+                            pageResult.Results.Add(new PublishedNode {
+                                NodeId = item.NodeId,
+                                PublishingInterval = item.PublishingInterval,
+                                SampligInterval = item.SamplingInterval,
+                            });
+                        }
+                    }
+                } while (!string.IsNullOrEmpty(continuationToken));
+            }
+            catch (Exception e) {
+                // skip this node
+                Trace.TraceError("Cannot get published nodes for endpointId'{0}'", endpointId);
+                var errorMessage = string.Format(e.Message, e.InnerException?.Message ?? "--", e?.StackTrace ?? "--");
+                Trace.TraceError(errorMessage);
+            }
+            pageResult.PageSize = 10;
```

Comment:
> 10; [](start = 33, length = 4)

again the magic number 10 ;-)

Reply:
added const value

Reply:
I don't see a difference from the previous commit?

---
In reply to: [356017686](https://github.com/Azure/Industrial-IoT/pull/167#discussion_r356017686) [](ancestors = 356017686)

Reply:
Maybe for the next version you can add an 'IPageConfig' which you define with specific parameters how to display rows/columns

---
In reply to: [355717729](https://github.com/Azure/Industrial-IoT/pull/167#discussion_r355717729) [](ancestors = 355717729)